lua/packer.lua

```diff
@@ -246,6 +246,7 @@ packer.install = function(...)
   end
 
   async(function()
+    vim.cmd 'PackerClean'
```

Comment:
This is a problem line. I've tried a lot of things and ultimately this is the only solution I could come across.

Here's what I _think_ is happening: whenever I replace `vim.cmd` with a direct call to `packer.clean()`, it uses an `async()` method. This means that the clean will occur after the install function tries to install the plugin, meaning that the directory has not been cleaned and the install always fails. I believe `vim.cmd` syncs the commands here.

I have tried a few things to get it to sync up without using a vim command:

1. `a.wait(packer.clean())`, though this causes only the clean to be performed and the install instructions thereafter are dropped. I'm not sure why this happens, although I'm guessing this has something to do with the Lua implementation of `async` not working exactly as one would expect async to work (since it is built on top of loosely scheduled coroutines, from what I can tell).
2. I have tried moving the `packer.clean()` call from out of this `async(function()` scope, but there's no difference to be made there.
3. I accidentally tried `async(packer.clean())`, and interestingly it gave the same result as `await(packer.clean())`

Maybe I'm missing something with the async implementation? Or maybe I'm barking up the wrong tree altogether. I feel like there's something obvious here but I can't quite see it…

Reply:
Ah, yes - sorry, the async framing has a bit of a learning curve (though I assert it makes things easier to write in net).

What you probably want here is either `await(clean(plugins, results))` or changing `packer.clean` to return an `async` function and using `await(packer.clean())`. The issue right now is that `packer.clean()` simply creates and immediately runs an `async` function (necessary because the `clean` function uses async operations and you cannot use `coroutine.yield` across a C function call), and returns nothing.